core/blockchain.go

```diff
@@ -2964,3 +2965,74 @@ func (bc *BlockChain) GetECDSAFromCoinbase(header *block.Header) (common.Address
 	}
 	return common.Address{}, ctxerror.New("cannot find corresponding ECDSA Address", "coinbaseAddr", header.Coinbase())
 }
+
+// SuperCommitteeForNextEpoch ...
+// TODO: remove shardID in parameter if not needed in worker code
+func (bc *BlockChain) SuperCommitteeForNextEpoch(
+	shardID uint32,
+	beacon consensus_engine.ChainReader,
+	header *block.Header,
+) (*shard.State, error) {
+	var (
+		nextCommittee = new(shard.State)
+		err           error
+	)
+	switch shardID {
+	case shard.BeaconChainShardID:
+		if shard.Schedule.IsLastBlock(header.Number().Uint64()) {
+			nextCommittee, err = committee.WithStakingEnabled.Compute(
+				new(big.Int).Add(header.Epoch(), common.Big1),
+				beacon,
+			)
+		}
+	default:
+		// TODO: needs to make sure beacon chain sync works.
+		beaconEpoch := beacon.CurrentHeader().Epoch()
```

Comment:
When this is called as verification instead of proposing. It shouldn't read from the CurrentHeader of beacon because it will change based on different timing. It should use the shardState.Epoch as beaconEpoch. In the case of before-staking, the shardState.epoch is nil, and in that case, the beaconEpoch is not needed for verification either.